.github/workflows/wasm.yml

```diff
@@ -0,0 +1,129 @@
+        shell: bash
+        id: ccache_vars
+        run: |-
+          echo "::set-output name=hash::$(echo ${{ env.ccache_compilercheck }})"
+          echo "::set-output name=timestamp::$(date '+%Y-%m-%dT%H.%M.%S')"
+
+      - name: Verify Emscripten setup
+        run: |
+            emcc --version
+            emcmake cmake --version
+            emmake make --version
+
+      - name: Set ccache environment for emcc
+        run: |
+          echo "CCACHE_COMPILER_CHECK=${{ env.ccache_compilercheck }}" >> $GITHUB_ENV
+          echo "CCACHE_BASEDIR=${{ env.ccache_basedir }}" >> $GITHUB_ENV
+          echo "CCACHE_COMPRESS=${{ env.ccache_compress }}" >> $GITHUB_ENV
+          echo "CCACHE_COMPRESSLEVEL=${{ env.ccache_compresslevel }}" >> $GITHUB_ENV
+          echo "CCACHE_DIR=${{ env.ccache_dir }}" >> $GITHUB_ENV
+          echo "CCACHE_MAXSIZE=${{ env.ccache_maxsize }}" >> $GITHUB_ENV
+          # https://emscripten.org/docs/compiling/Building-Projects.html#using-a-compiler-wrapper
+          echo "EM_COMPILER_WRAPPER=ccache" >> $GITHUB_ENV
+          
+
+      - name: Cache-op for build-cache through ccache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.ccache_dir }}
+          key: ccache-${{ job.id }}-${{ steps.ccache_vars.outputs.hash }}-${{ github.ref }}-${{ steps.ccache_vars.outputs.timestamp }}
+          restore-keys: |-
+            ccache-${{ job.id }}-${{ steps.ccache_vars.outputs.hash }}-${{ github.ref }}
+            ccache-${{ job.id }}-${{ steps.ccache_vars.outputs.hash }}
+            ccache-${{ job.id }}
+
+      - name: Configure builds
+        run: |
+          mkdir -p build-wasm
+          cd build-wasm
+          emcmake cmake -DCOMPILE_WASM=on ..
+
+      - name: ccache prolog
+        run: |-
+          ccache -s # Print current cache stats
+          ccache -z # Zero cache entry
+
+      - name: Compile
+        working-directory: build-wasm
+        run: |
+          emmake make -j2
+
+      - name: ccache epilog
+        run: |
+          ccache -s # Print current cache stats
+
+      - name: Instantiate simd wormhole
+        working-directory: build-wasm
+        run: bash ../wasm/patch-artifacts-enable-wormhole.sh
+
+      - name: Import GEMM library from a separate wasm module
+        working-directory: build-wasm
+        run: bash ../wasm/patch-artifacts-import-gemm-module.sh
+
+      - name: Check artifacts
+        working-directory: build-wasm
+        run: |
+          ls -all bergamot*
+          if ls bergamot*.wasm &>/dev/null && ls bergamot*.js &>/dev/null
+          then
+            echo "Artifacts Successfully Generated"
+          else
+            echo "Failure: Artifacts Not Present"
+            exit 1
+          fi
+
+      - name: Upload wasm artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: bergamot-translator-worker
+          path: ${{github.workspace}}/build-wasm/bergamot-translator-worker.*
```

Comment:
Wouldn't it make sense to have the "Check Artifacts" step check the same paths as the ones you upload? Or better, you can skip the "Check artifacts" step and [make the upload artifacts step fail if no files are found](https://github.com/actions/upload-artifact#customization-if-no-files-are-found).

Reply:
I need to get it to `$GITHUB_PATH` (manually) or use source script in every shell, I think given the existing setting? I like the idea. The example pointed to is JS for consumption in actions, unsure how to wire the concept in here. Another angle is I can watch pyiodide (which seems to be gaining some traction) and replicate what they're doing.